python/pyhail/context.py

```diff
@@ -0,0 +1,457 @@
+
+from pyhail.dataset import VariantDataset
+from pyhail.java import jarray, scala_object
+
+class HailContext:
+    """:class:`.HailContext` is the main entrypoint for PyHail
+    functionality.
+    
+    :param SparkContext sc: The pyspark context.
+    """
+    
+    def __init__(self, sc):
+        self.sc = sc
+        
+        self.gateway = sc._gateway
+        self.jvm = sc._jvm
+        
+        # sc._jsc is JavaObject JavaSparkContext
+        self.jsc = sc._jsc.sc()
+        
+        self.jsqlContext = sc._jvm.SQLContext(self.jsc)
+        
+        self.sqlContext = pyspark.sql.SQLContext(sc, self.jsqlContext)
+        
+        self.jsc.hadoopConfiguration().set(
+            'io.compression.codecs',
+            'org.apache.hadoop.io.compress.DefaultCodec,org.broadinstitute.hail.io.compress.BGzipCodec,org.apache.hadoop.io.compress.GzipCodec')
+        
+        logger = sc._jvm.org.apache.log4j
+        logger.LogManager.getLogger("org"). setLevel(logger.Level.ERROR)
+        logger.LogManager.getLogger("akka").setLevel(logger.Level.ERROR)
+    
+    def _jstate(self, jvds):
+        return self.jvm.org.broadinstitute.hail.driver.State(
+            self.jsc, self.jsqlContext, jvds, scala_object(self.jvm.scala.collection.immutable, 'Map').empty())
+    
+    def _run_command(self, vds, pargs):
+        jargs = jarray(self.gateway, self.jvm.java.lang.String, pargs)
+        t = self.jvm.org.broadinstitute.hail.driver.ToplevelCommands.lookup(jargs)
+        cmd = t._1()
+        cmd_args = t._2()
+        result = cmd.run(self._jstate(vds.jvds if vds != None else None),
+                         cmd_args)
+        return VariantDataset(self, result.vds())
+    
+    def fam_summary(input, output):
+        """Outputs summary of a .fam file.
+        
+        :param str input: Input .fam file.
+        
+        :param str output: Output summary file.
+        
+        :return: Nothing.
+
+        """
+        pargs = ["famsummary", "-f", input, "-o", output]
+        _run_command(self, pargs)
+
+    def grep(regex, path, max_count = 100):
+        """Grep big files, like, really fast.
+        
+        :param str regex: The regular expression to match.
+        
+        :param path: The files to search.
+        :type path: str or list of str
+
+        :param int max_count: The maximum number of matches to return.
+
+        :return: Nothing.
+        """
+        
+        pargs = ["grep", regex]
+        if isinstance(path, str):
+            pargs.append(path)
+        else:
+            for p in path:
+                pargs.append(p)
+        
+        pargs.append('--max-count')
+        pargs.append(str(max_count))
+        
+        _run_command(self, pargs)
+    
+    def import_annotations_table(self, path, variant_expr, code = None, npartitions = None,
+                                 # text table options
+                                 types = None, missing = "NA", delimiter = "\\t", comment = None,
```

Comment:
need to decide on python style, whether to do this indentation or one tab like we do in scala